### PR TITLE
Cleanup SPMTestSupport Module

### PR DESCRIPTION
**Pull request by mattt**
One last incremental PR in service of #2967, this time doing some clean up in the SPMTestSupport module.

In my work to update the semantics of Swift package identity, I've spent a lot of time in `Workspace`-adjacent tests. And one of the things that's been tripping me up is the use of `Test___` as a prefix for supporting types, like `TestTarget` and `TestProduct`, which I often conflate with `.testTarget` or the like. I think the prefix `Mock___` provides clearer, more descriptive names for these types.

Along the way, I also fixed some unused or misnamed APIs that I encountered. For example, there was an unused `MockDependencyGraph` type in addition to a `MockGraph` type that was used, so I renamed the latter to the former  (2f283f09a2ab1ed3e5c8e6db7a06e11f12dc641f).

As always, most of these changes are severable from one another and can be picked or dropped a la carte.

<ins>Edit</ins>: Any incidental formatting or whitespace changes are from SwiftFormat (#3012), which I have set up to run automatically in a `pre-commit` hook (`.git/hooks/pre-commit`) for each staged file:

```bash
#!/bin/bash
git-format-staged --formatter "swiftformat stdin --stdinpath '{}'" "*.swift"
```

### CODE REVIEW

**Comment by mattt** [comment]
@swift-ci please smoke test

**Comment by mattt** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
Thanks for all the continuing cleanup, @mattt!  I agree about the change from Test to Mock, but just had a couple of questions about the other changes.

File: Sources/SPMTestSupport/MockWorkspace.swift
Comment (mattt) [reply Sources/SPMTestSupport/MockWorkspace.swift:62]: These were automatically inserted by SwiftFormat, for which @tomerd added the configuration with #3012. I believe  we share stylistic preferences, but I'm more than happy to accede to any conventions enforced by a formatter.

That said, I've encountered a few situations in which SwiftFormat makes source-breaking changes to Swift Package Manager. Namely, mistakenly inserting `self.` for things like `break` labels and local variables whose names shadow an instance method or property. I'm working to share minimal, reproducible examples with the maintainer, but this may be cause for reconsidering how automatic formatting is currently set up.

File: Tests/XCBuildSupportTests/PIFBuilderTests.swift
Comment (mattt) [reply Tests/XCBuildSupportTests/PIFBuilderTests.swift:141]: This is a consequence of a3a67f960d5311614c378864651f8ea09162fac0, which removes a helper overload that implies the `type` parameter. I found the ambiguity at the call site not to be worth the benefit of the affordance. If it makes sense for this to be the default argument, I think that change should be made in the original declaration.

File: Sources/SPMTestSupport/MockWorkspace.swift
Comment (tomerd) [reply Sources/SPMTestSupport/MockWorkspace.swift:62]: > Are all these self. additions just stylistic? I think the general Swift approach (and the one used in SwiftPM) is to not specify self. when it's not needed (such as in an escaping closure).

> These were automatically inserted by SwiftFormat, for which @tomerd added the configuration with #3012. I believe we share stylistic preferences, but I'm more than happy to accede to any conventions enforced by a formatter.

In other projects (SwiftNIO, etc) we actually encouraged adding self. to make sure things are very explicit. happy to open up for discussion tho. cc @weissi

> That said, I've encountered a few situations in which SwiftFormat makes source-breaking changes to Swift Package Manager. Namely, mistakenly inserting self. for things like break labels and local variables whose names shadow an instance method or property. 

interesting. I have not had such trouble with SwiftFormat in other projects, what version are you using @mattt?

Comment (tomerd) [reply Sources/SPMTestSupport/MockWorkspace.swift:62]: @mattt in any case dont let this block progress. you can run the formatter with the `self` setting if it breaks things and we can revisit this separately

Comment (mattt) [reply Sources/SPMTestSupport/MockWorkspace.swift:62]: @tomerd According to `swiftformat --version`, I'm using the latest (0.47.2). For now, I'm committing with `--no-verify` to skip my pre-commit hook as necessary. So this isn't a blocker.

File: Sources/SPMTestSupport/XCTAssertHelpers.swift
Comment (mattt) [reply Sources/SPMTestSupport/XCTAssertHelpers.swift:102]: In the interest of keeping this focused, let's punt on that. For something like `#file` usage, I think we'll want to change everything project-wide at once, rather than adopting piecemeal over time.

**Comment by mattt** [comment]
@abertelrud Thanks so much for reviewing this. I just finished responding to all of your feedback. Let me know if there's anything else you'd like to discuss before merging this in.

File: Sources/SPMTestSupport/MockWorkspace.swift
Comment (abertelrud) [reply Sources/SPMTestSupport/MockWorkspace.swift:62]: This is fine, it's just opposite of what I've seen before so I wanted to ask.

File: Sources/SPMTestSupport/XCTAssertHelpers.swift
Comment (abertelrud) [reply Sources/SPMTestSupport/XCTAssertHelpers.swift:102]: Sounds good.

File: Tests/XCBuildSupportTests/PIFBuilderTests.swift
Comment (abertelrud) [reply Tests/XCBuildSupportTests/PIFBuilderTests.swift:141]: Thanks for clarifying.

**Comment by abertelrud** [comment]
Thanks, this looks great.  FYI, I also have a separate change to make SPMTestSupport able to be used for libSwiftPMDataModel, but we can merge yours first and I can rebase on that.

**Review by abertelrud — APPROVED** [review]
Is this ready to merge?

**Comment by abertelrud** [comment]
Actually, based on your previous comment it sounds as if it is, so I'll merge.

**Pull request merged**
